convert_hf_to_gguf.py

```diff
@@ -94,6 +108,7 @@ def __init__(self, dir_model: Path, ftype: gguf.LlamaFileType, fname_out: Path,
         self.dir_model = dir_model
         self.ftype = ftype
         self.fname_out = fname_out
+        self.hf_arch = hf_arch
```

Comment:
FYI, there is `get_model_architecture` that accepts `hparams`, maybe you can reuse it? (and may put it inside `TextModel` for now)

Reply:
get_model_architecture is typed to expect a non-null path. Even extracting that, it expects a ModelType, but TextModel and VisionModel cannot actually be mapped back to that - they would need something like this:
```python
class TextModel(ModelBase):
    model_type = ModelType.TEXT
    # ...

class VisionModel(ModelBase):
    model_type = ModelType.VISION
    # ...
```

Is it not simpler to just let the model know the arch it's being constructed for?

Reply:
Yes but I think it can be read automatically from `hparams`. Or, can you patch `get_model_architecture` so that it accepts either `hparams` or `path`?

I should have done this earlier because some vision models relies on the `hf_arch`

Reply:
adding `self.model_type` is also fine IMO

Reply:
Better now?